### DISCUSSION TITLE
How to convert plotnine object to matplotlib object?

### DISCUSSION THREAD

**Discussion by ericjb** [has2k1/plotnine#755] [Q&A] [answered]
The package sktime seeks to extend sklearn so that time series analysis can easily incorporate machine learning pipelines, machine learning algorithms etc. sktime has some convenient graphics utilities (i.e. plotting) which can return (or modify) matplotlib figure and axes objects. The sktime graphics implementation is a bit buggy, and I am investigating ripping out the matplotlib calls and replacing them with plotnine calls. But I would still like to return matplotlib objects so as not to break existing code that uses the sktime plotting routines. I tried ggplot(..).draw() but am having problems. Any pointers would be greatly appreciated.

**Comment by has2k1** [comment, marked as answer]
`ggplot().draw()` returns a Matplotlib figure. From the figure, you can get the axes. The figure has a custom layout manager that can handle the artists plotnine adds to it. So should not be changed.

**Comment by j-craggy** [comment]
Now how to convert matplotlib.pyplot into plotnine.ggplot?
